### PR TITLE
Adds a cringe word filter, saying something cringe IC makes you receive a mood debuff

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

TLDR using words like "redditor", "chungus", "pewdiepie", "based", etc IC will make you suffer a big mood debuff.
image
image
image

**UPDATE:** Based now checks for the words before and after it. Saying stuff like "Based on..." or "Analitically based..." will not give you the debuff, but saying "This is based."  or "Based." will.

## Why It's Good For The Game

This is a better solution than staff cracking down on LRP behavior that nudges people just a bit into acting mature.

## Changelog
:cl:
add: Added an IC word filter - saying internet slang like "based" and "cringe" will send you a very negative moodlet
/:cl:

### CODE REVIEW

**Comment by Owai-Seek** [comment]
I somewhat like the idea, but there's some words in this list that I feel have a place/situations where they're often used.

Specifically, "cuck", "banned" "grief" "IC" and "OOC"

OOC specifically is used when you need to ask someone a somewhat private OOC question, but there's no whisper/subtle for LOOC. IE: (OOC: I'm not sure what X is, could you elaborate?) or (OOC: I think you misunderstood, I did X, not Y)

I could elaborate on the other ones, but I don't want to make a longwinded post about it at the moment.

**Comment by BobJoga** [comment]
> I somewhat like the idea, but there's some words in this list that I feel have a place/situations where they're often used.
> 
> Specifically, "cuck", "banned" "grief" "IC" and "OOC"
> 
> OOC specifically is used when you need to ask someone a somewhat private OOC question, but there's no whisper/subtle for LOOC. IE: (OOC: I'm not sure what X is, could you elaborate?) or (OOC: I think you misunderstood, I did X, not Y)
> 
> I could elaborate on the other ones, but I don't want to make a longwinded post about it at the moment.

Keep LOOC on and this problem is completely avoided. Saying OOC and IC and other meta phrases ingame should not in any way be accepted.

**Comment by Owai-Seek** [comment]
The point is, you're using subtle or whispering in a public location, but don't want to use LOOC because it blasts it to everyone in the area. There's no subtle/whisper for LOOC messages. (1 tile around) so people usually subtle or whisper that something is OOC. I've VERY RARELY seen OOC/IC used in-character, and those people usually learn once they're told not to do it.

**Comment by IrkallaEpsilon** [comment]
I kinda like it, though "grief" or "banned" have alot of legitimate use. "Smoking in my maintenance bar is now banned." and "Good grief." for instance

**Comment by davidchan13** [comment]
Why the fuck is Obama a cringe word?

Should really be focusing on meta words; emag, malf, ling and shit like that.

**Comment by Yawet330** [comment]
daily reminder braindamage will cause you to become suicidal if this PR gets merged, some of the trauma phrases causes you to scream _multiple_ of these at once.

**Comment by BobJoga** [comment]
> I kinda like it, though "grief" or "banned" have alot of legitimate use. "Smoking in my maintenance bar is now banned." and "Good grief." for instance

I have removed those now 👍

**Comment by Seris02** [comment]
Same with "Based"
as much as I don't like it, maybe make it only when said alone?
because something like "based on bla" is something that can be said
perhaps make just "based" by itself only do that?
as in Bla says, "Based." 
would give you the debuff but
Bla says, "Based on the data, bla bla bla" would not give you it.

**Comment by MemedHams** [comment]
"**Based** on our recent data..."
"I'm going to make an executive decision **based** on my authority as captain..."
"**Based** on the evidence against you, I'm going to have to put you in perma..."
Based is a real word with a variety of uses outside of cringe culture, please do not make the mistake of banning a multipurpose insert like this, Peniston and Scunthorpe did not suffer so we could ignore the lesson in targeted word bans they put forth. Also, despite my reservations on the matter, wtf is something I'd consider to be pretty in char, as I've heard it said in real life a good number of times, outside of shitty internet memes.

**Comment by MemedHams** [comment]
Also, how are instances of this being measured? "ic", "pog", and "nae" can all exist as a segment of an existing word. Additionally, this entire soft wordblock can be circumvented by simply saying "I C" or "p o g" or something of that variety. (Oh god please do not take this advice the wrong way, if you just say "oh okay I'll just add those then" you'll open a whole new can of worms of accidentally beaning people for just happening to have those spaced letters as part of something else)

**Comment by BobJoga** [comment]
> Also, how are instances of this being measured? "ic", "pog", and "nae" can all exist as a segment of an existing word. Additionally, this entire soft wordblock can be circumvented by simply saying "I C" or "p o g" or something of that variety. (Oh god please do not take this advice the wrong way, if you just say "oh okay I'll just add those then" you'll open a whole new can of worms of accidentally beaning people for just happening to have those spaced letters as part of something else)

This only measures for the full word, to avoid this kind of conflict. Yes, this can be circumented, but if you go as far as to go out of your way to avoid the mood penalty from this, admins should intervene.

**Comment by BobJoga** [comment]
> "**Based** on our recent experiments with slime crossbreeds..."
> "I'm going to make an executive decision **based** on my authority as captain..."
> "**Based** on the evidence against you, I'm going to have to put you in perma..."
> Based is a real word with a variety of uses outside of cringe culture, please do not make the mistake of banning a multipurpose insert like this, Peniston and Scunthorpe did not suffer so we could ignore the lesson in targeted word bans they put forth. Also, despite my reservations on the matter, wtf is something I'd consider to be pretty in char, as I've heard it said in real life a good number of times, outside of shitty internet memes.

Wtf is an abbreviation and you sound really dumb saying it out loud. That said, i rarely see the word based being used in-game for real reasons besides based cringe kek and that kind of stuff.

**Comment by BobJoga** [comment]
> Same with "Based"
> as much as I don't like it, maybe make it only when said alone?
> because something like "based on bla" is something that can be said
> perhaps make just "based" by itself only do that?
> as in Bla says, "Based."
> would give you the debuff but
> Bla says, "Based on the data, bla bla bla" would not give you it.

Quite hard to do if i'm quite honest.

**Comment by MemedHams** [comment]
still trouble with based- I can understand trying to prevent abuse, but it is a real word with real word, inoffensive applications.

**Comment by ThatGuyFromThatPlace** [comment]
Based is a word that is used frequently. I'd say only include if it is possible to include it if it's said by itself.
Also, I think, removing the *dab is more important than punishing people for saying the word

**Comment by NopemanMcHalt** [comment]
Absolutely **not our man** and **normal-minded**.
/s

**Comment by NotRanged** [comment]
I suggest a testmerge before fullmerge considering how often speech and emote procs are called, adding a whole bunch of new loops and string comparisons might be a peformance concern?

Though the mood system is total garbage anyway imo, I always have the quirk that hides that shit from me (and I get points for it? Sweet.)

I also don't think this is the way to go about this at all, perhaps an admin niche log when someone says some dumb LRP shit would be better, let them decide the punishment and crack down on it. There's also quite a few legitimate uses for some of the words in the list still, it should really be handled on a per-case basis.

**Comment by BobJoga** [comment]
> I suggest a testmerge before fullmerge considering how often speech and emote procs are called, adding a whole bunch of new loops and string comparisons might be a peformance concern?
> 
> Though the mood system is total garbage anyway imo, I always have the quirk that hides that shit from me (and I get points for it? Sweet.)
> 
> I also don't think this is the way to go about this at all, **perhaps an admin niche log when someone says some dumb LRP shit would be better**, let them decide the punishment and crack down on it. There's also a few legitimate uses for some of the words in the list still, it should really be handled on a per-case basis.

I don't trust admins, never will - If a system relies on OOC intervention then it is bad and i'd rather kill this idea. I don't have anything else to say about that.

**Comment by BobJoga** [comment]
> Based is a word that is used frequently. I'd say only include if it is possible to include it if it's said by itself.
> Also, I think, removing the *dab is more important than punishing people for saying the word. What if I'm saying "This person just did something stupid then dabbed"

I never see dabbed being used outside of meme reasons, but based is common and i'll have to find a solution.

**Comment by Zenitheevee** [comment]
Hmmm... The only word i see _Potentially_ (And i have to say Potentially because its a word nearly no one fucking uses) Having a problem is "Dab", But thats because it is used for a small, gentle, spot-focused action (dabbing on some sterilizer on a small puncture, or dabbing some paint on a canvas), However that would imply people would actually start to use it for descriptives, Which would mean the whole dab family. (But again, Nearly no one fucking uses it :v)

And, While i cant say its foolproof, it _**IS**_ a good start. Would there be more additions i could name off the back of my head? Well, outside of the word "Gamer" (And even then, i'm kinda straddling the fence on it), and potentially antag-related meme words (Ling, Tator, Vampies, ETC ETC, and this would mostly be so people will.. You know... Say the whole fucking word), Not much i could personally add.

**Comment by BobJoga** [comment]
> Same with "Based"
> as much as I don't like it, maybe make it only when said alone?
> because something like "based on bla" is something that can be said
> perhaps make just "based" by itself only do that?
> as in Bla says, "Based."
> would give you the debuff but
> Bla says, "Based on the data, bla bla bla" would not give you it.

Ok, i've updated it - Now only saying based alone will give you the debuff.

**Comment by YakumoChen** [comment]
Honestly why not just port TG's word blacklist instead of this? This PR itself is pretty cringe. If you want to stop OOC in IC, actually do it, don't just beat around the bush.

**Comment by BobJoga** [comment]
> Honestly why not just port TG's word blacklist instead of this? This PR itself is pretty cringe. If you want to stop OOC in IC, actually do it, don't just beat around the bush.
> 
> Also we have dabbing as an emote, why is it a cringe word?

It's a bad emote and should probably be removed, but i think making people receive a mood debuff for doing it is enough. I also need the word blacklist PR in question.

File: modular_skyrat/code/datums/mood_events/generic_negative_events.dm [hunk 1]
```diff
@@ -3,3 +3,9 @@
 	description = "<span class='userdanger'>WHERE IS THE EPI PEN?!?!</span>\n"
 	mood_change = -10
 	timeout = 10 SECONDS
+
+//For when you say something really awful IC.
+/datum/mood_event/cringe
+	description = "<span class='boldwarning'>God, they're gonna put me in a cringe compilation after that...</span>\n"
```
Comment (H0gberg) [inline modular_skyrat/code/datums/mood_events/generic_negative_events.dm:9]: I feel like if it said something a bit more grounded like, "did I just said that out loud? God, that's embarrassing..." would go a long way, cause it saying "cringe comp" seems to be adding to the issue which is to stamp out the memes

Comment (BobJoga) [reply modular_skyrat/code/datums/mood_events/generic_negative_events.dm:9]: doing it
